src/legacy/fetch.ts

```diff
@@ -75,12 +75,13 @@ export const checkedFetch = (request: FetchRequest) => {
       return response
     })
 }
-export const fetchDefault = (
+export const fetchDefault = <T extends typeof _checkResponse>(
   path: string,
   payload: any,
   networkConfig: BackendConfig,
   method: RequestMethod = 'POST',
-) => {
+  options?: {checkResponse?: T},
```

Comment:
I'd rather receive a type guard here than hack typing.

Reply:
Does the following work for you? I think we may need a parser instead of a type guard. Please let me know what are your thoughts about the following.

```
  type ResponseChecker<T> = (rawResponse: Record<string, any>, requestPayload: Record<string, any>) => Promise<T>

```